server/src/main/java/org/apache/iotdb/db/service/basic/ServiceProvider.java

```diff
@@ -206,21 +206,26 @@ public BasicOpenSessionResp login(
       openSessionResp.setCode(TSStatusCode.SUCCESS_STATUS.getStatusCode());
       openSessionResp.setMessage("Login successfully");
 
-      SESSION_MANAGER.supplySession(session, username, zoneId, clientVersion);
-      LOGGER.info(
-          "{}: Login status: {}. User : {}, opens Session-{}",
-          IoTDBConstant.GLOBAL_DB_NAME,
-          openSessionResp.getMessage(),
-          username,
-          session);
+      SESSION_MANAGER.supplySession(session, username, zoneId, clientVersion, enableAudit);
+      if (!AuditLogUtils.LOG_LEVEL_NONE.equals(CONFIG.getAuditLogStorage())) {
+        AuditLogUtils.writeAuditLog(
+            String.format(
+                "%s: Login status: %s. User : %s, opens Session-%s",
+                IoTDBConstant.GLOBAL_DB_NAME, openSessionResp.getMessage(), username, session));
+      }
+
     } else {
       openSessionResp.setMessage(loginMessage != null ? loginMessage : "Authentication failed.");
       openSessionResp.setCode(TSStatusCode.WRONG_LOGIN_PASSWORD_ERROR.getStatusCode());
-      AUDIT_LOGGER.info("User {} opens Session failed with an incorrect password", username);
+      session.setUsername(username);
+      if (!AuditLogUtils.LOG_LEVEL_NONE.equals(CONFIG.getAuditLogStorage())) {
+        AuditLogUtils.writeAuditLog(
+            String.format("User %s opens Session failed with an incorrect password", username),
+            true);
```

Comment:
It means that we need to record all the failed login info?

Reply:
yes, the user wants to know which illegitimate users are connected to the database